managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ReadOnlyCursor.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.mledger;
+
+import java.util.List;
+
+import org.apache.bookkeeper.mledger.AsyncCallbacks.ReadEntriesCallback;
+
+public interface ReadOnlyCursor {
```

Comment:
Since this is a subset of ManagedCursor functionallity, ManagedCursor should extends this interface

Reply:
There is only a small difference in the interface. `skipEntries()` for the `ManagedCursor` is a blocking operation since it will move (and persist) the mark-delete position. For `ReadOnlyCursor`, there is no notion of deletion, so the skip is just referring to moving the read position ahead, which is a local operation.